### PR TITLE
[REF] sale: settings by company

### PR DESCRIPTION
**Pull request by Feyensv**
Enterprise PR: https://github.com/odoo/enterprise/pull/30168
Upgrade PR: https://github.com/odoo/upgrade/pull/3965

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/wizard/res_config_settings.py [hunk 1]
```diff
@@ -82,11 +75,6 @@ class ResConfigSettings(models.TransientModel):
 
     #=== ONCHANGE METHODS ===#
 
-    @api.onchange('use_quotation_validity_days')
-    def _onchange_use_quotation_validity_days(self):
-        if self.quotation_validity_days <= 0:
-            self.quotation_validity_days = self.env['res.company'].default_get(['quotation_validity_days'])['quotation_validity_days']
-
```
Comment (Demesmaeker) [inline addons/sale/wizard/res_config_settings.py:89]: L80: `if self.quotation_validity_days < 0:` ?
And adapt the warning message?
Right now, we can't set the default quotation validity to 0 in the setting, which is quite sad

**Review by Demesmaeker — COMMENTED** [review]
:duck:

File: addons/sale/data/ir_config_parameter.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo noupdate="1">
+
+    <record id="default_confirmation_template" model="ir.config_parameter">
+        <field name="key">sale.default_confirmation_template</field>
+        <field name="value" ref="sale.mail_template_sale_confirmation"/>
+    </record>
```
[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/sale/data/ir_config_parameter.xml:7]: Yeah, kinda postponed the decision, but let's fight and revert the commit that stopped considering the parameter content.

[the remark this answers is not in the document]
Comment (Demesmaeker) [reply addons/sale/data/ir_config_parameter.xml:7]: Yup, just saw it, indeed, it's more logical =]

**Review by Demesmaeker — APPROVED** [review]

**Comment by Feyensv** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**